### PR TITLE
DACT-2279: Nationality formatting not being checked correctly

### PR DESCRIPTION
**Pull request by cdrainCH**
JIRA: https://companieshouse.atlassian.net/browse/DACT-2279
Fixes the formatting of nationalities to reflect the drop down list

### CODE REVIEW

File: src/controllers/active.directors.controller.ts [hunk 1]
```diff
@@ -162,9 +163,9 @@ async function beginUpdateJourney(req: Request, res: Response, session: Session,
     dateOfBirth: appointment.dateOfBirth?.year + "-" + appointment.dateOfBirth?.month?.toString().padStart(2, '0') + "-" + appointment.dateOfBirth?.day?.toString().padStart(2, '0'),
     appointedOn: appointment.appointedOn? appointment.appointedOn: appointment.appointedBefore,
     occupation: occupation,
-    nationality1: nationalities && nationalities?.length > 0? formatTitleCase(nationalities[0]): "",
-    nationality2: nationalities && nationalities?.length > 1? formatTitleCase(nationalities[1]): "",
-    nationality3: nationalities && nationalities?.length > 2? formatTitleCase(nationalities[2]): "",
+    nationality1: nationalities && nationalities?.length > 0? lowerCaseAllWordsExceptFirstLetters(nationalities[0]): "",
```
Comment (AndreKramer7654) [inline src/controllers/active.directors.controller.ts:166]: You might be able to drop the ? from nationalities?.length to get test coverage up. It's checked by first condition so the ? is not needed. For lines 166 to 168.

File: src/controllers/shared.controllers/director.nationality.controller.ts [hunk 2]
```diff
@@ -120,9 +120,15 @@ const sameNationalityWithChips = (currentOfficerFiling: OfficerFiling, companyAp
...
-    if ((!currentOfficerFiling.nationality2 || nationality[1] === currentOfficerFiling.nationality2) &&
-        (!currentOfficerFiling.nationality3 || nationality[2] === currentOfficerFiling.nationality3)) {
+  nationality.forEach((officerNationality, index)=> {
+    if (officerNationality) {
+      nationality[index] = officerNationality.toUpperCase();
+    }
+  });
+
+  if (currentOfficerFiling.nationality1?.toUpperCase() === nationality[0]) {
+    if ((!currentOfficerFiling.nationality2 || nationality[1] === currentOfficerFiling.nationality2.toUpperCase()) &&
```
Comment (jbestwickch) [inline src/controllers/shared.controllers/director.nationality.controller.ts:130]: Wouldn't this falsely flag nationalities being the same if nationality2 had been removed/blanked in the filing but was present on CHIPS? In a scenario where dual nationality was being removed for instance

File: src/utils/format.ts [hunk 3]
```diff
@@ -123,3 +123,27 @@ export const retrieveDirectorNameFromFiling = (filing: OfficerFiling ): string =
     return "";
   }
 }
+
+export const lowerCaseAllWordsExceptFirstLetters = (country: string | undefined) => {
+  if (!country){
+    return "";
+  }
+
+  const wordsForAllLowerCase = ["AND", "OF", "THE", "DA", "PART", "CITIZEN"];
```
Comment (jbestwickch) [inline src/utils/format.ts:132]: It feels like this should be config, may need to be expanded in the future if new countries are added or renamed.

File: src/utils/format.ts [hunk 4]
```diff
@@ -123,3 +123,27 @@ export const retrieveDirectorNameFromFiling = (filing: OfficerFiling ): string =
     return "";
   }
 }
+
+export const lowerCaseAllWordsExceptFirstLetters = (country: string | undefined) => {
```
[the remark this answers is not in the document]
Comment (jbestwickch) [reply src/utils/format.ts:127]: Also I think this formatting style is know as sentence case.

[the remark this answers is not in the document]
Comment (cdrainCH) [reply src/utils/format.ts]: Name of method and parameter changed

File: src/controllers/active.directors.controller.ts (hunk 1 above)
Comment (cdrainCH) [reply src/controllers/active.directors.controller.ts]: ? removed, test coverage has increased :D

File: src/controllers/active.directors.controller.ts [hunk 5]
```diff
@@ -162,9 +163,9 @@ async function beginUpdateJourney(req: Request, res: Response, session: Session,
     dateOfBirth: appointment.dateOfBirth?.year + "-" + appointment.dateOfBirth?.month?.toString().padStart(2, '0') + "-" + appointment.dateOfBirth?.day?.toString().padStart(2, '0'),
     appointedOn: appointment.appointedOn? appointment.appointedOn: appointment.appointedBefore,
     occupation: occupation,
-    nationality1: nationalities && nationalities?.length > 0? formatTitleCase(nationalities[0]): "",
-    nationality2: nationalities && nationalities?.length > 1? formatTitleCase(nationalities[1]): "",
-    nationality3: nationalities && nationalities?.length > 2? formatTitleCase(nationalities[2]): "",
+    nationality1: nationalities && nationalities.length > 0? formatNationalitiesToSentenceCase(nationalities[0]): "",
```
Comment (okoye-companieshouse) [inline src/controllers/active.directors.controller.ts:166]: nationality1: nationalities?.[0] ? formatNationalitiesToSentenceCase(nationalities[0]) : "",
nationality2: nationalities?.[1] ? formatNationalitiesToSentenceCase(nationalities[1]) : "",
nationality3: nationalities?.[2] ? formatNationalitiesToSentenceCase(nationalities[2]) : "",

File: src/utils/format.ts (hunk 3 above)
Comment (AndreKramer7654) [reply src/utils/format.ts:132]: Improving on what ROE used, go for it and move to global constants.

File: src/controllers/shared.controllers/director.nationality.controller.ts [hunk 6]
```diff
@@ -120,9 +120,15 @@ const sameNationalityWithChips = (currentOfficerFiling: OfficerFiling, companyAp
...
-  if (currentOfficerFiling.nationality1 === nationality[0]) {
-    if ((!currentOfficerFiling.nationality2 || nationality[1] === currentOfficerFiling.nationality2) &&
-        (!currentOfficerFiling.nationality3 || nationality[2] === currentOfficerFiling.nationality3)) {
+  nationality.forEach((officerNationality, index)=> {
+    if (officerNationality) {
+      nationality[index] = officerNationality.toUpperCase();
+    }
+  });
+
+  if (currentOfficerFiling.nationality1?.toUpperCase() === nationality[0]) {
```
Comment (AndreKramer7654) [inline src/controllers/shared.controllers/director.nationality.controller.ts:129]: Don't need the ? as there must be a nationality1. (for Coverage)

File: src/utils/format.ts (hunk 3 above)
Comment (cdrainCH) [reply src/utils/format.ts]: List moved to constants

File: src/controllers/shared.controllers/director.nationality.controller.ts (hunk 6 above)
Comment (AndreKramer7654) [reply src/controllers/shared.controllers/director.nationality.controller.ts:129]: But optional in filing.

**Review by PhilipGlover — APPROVED** [review]
👍

**Pull request merged**